### PR TITLE
Add ImportLocation to SymbolToken

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: IonDotnet/Tree/IonSymbol.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@ namespace IonDotnet.Tree
     public sealed class IonSymbol : IonText
     {
         private int _sid;
+        private ImportLocation _importLocation;
 
         public IonSymbol(string text, int sid = SymbolToken.UnknownSid) : this(new SymbolToken(text, sid))
```
Comment (byronlin13) [inline IonDotnet/Tree/IonSymbol.cs]: SymbolToken object creation is missing importLocation parameter

File: IonDotnet/Internals/SharedSymbolTable.cs [hunk 2]
```diff
@@ -110,7 +110,8 @@ public SymbolToken Find(string text)
                 return default;
 
             var internedText = _symbolNames[sid - 1];
-            return new SymbolToken(internedText, sid);
+            // Normalize the SID and import reference to this table.
+            return new SymbolToken(internedText, sid, new ImportLocation(Name, sid));
```
Comment (R-maan) [inline IonDotnet/Internals/SharedSymbolTable.cs]: Can we refactor this constructor so we do not need to pass "sid" twice? (once as the second param and once within the ImportLocation object)

File: IonDotnet/Tree/IonSymbol.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ namespace IonDotnet.Tree
     public sealed class IonSymbol : IonText
     {
         private int _sid;
+        private ImportLocation _importLocation;
```
Comment (R-maan) [inline IonDotnet/Tree/IonSymbol.cs:8]: If ``_importLocation.Sid`` is the same as ``_sid`` and always present, can we eliminate ``_sid`` and use ``_importLocation.Sid`` instead?

Comment (byronlin13) [reply IonDotnet/Tree/IonSymbol.cs:8]: According to Avneet:

> Symbols will only have an import location if they were declared in a shared symbol table. And symbols with an import location will only have a local SID in the context of a local symbol table. Note that the developer guide does not actually require the local SID, although this is potentially a performant way to cache the value when it is known that the symbol will be written in the same local symbol table context.

File: IonDotnet/Tree/IonSymbol.cs (hunk 1 above)
Comment (ghost) [reply IonDotnet/Tree/IonSymbol.cs]: Added.

File: IonDotnet.Tests/Internals/SymbolTokenTest.cs [hunk 4]
```diff
@@ -58,5 +98,6 @@ public void ArrayResize_RemainderSetToDefault(int oldLength, int newLength)
         }
 
         private static SymbolToken CreateSampleToken() => new SymbolToken("yo", 30);
+        private static SymbolToken CreateSampleTokenWihtImportLocation() => new SymbolToken("yo", 30, new ImportLocation("hey", 40));
```
Comment (byronlin13) [inline IonDotnet.Tests/Internals/SymbolTokenTest.cs]: With*

**Pull request closed without merging**